swarm/storage/ldbstore.go

```diff
@@ -659,38 +761,26 @@ func (s *LDBStore) writeCurrentBatch() error {
 	if l == 0 {
 		return nil
 	}
-	e := s.entryCnt
-	d := s.dataIdx
-	a := s.accessCnt
 	s.batch = newBatch()
-	b.err = s.writeBatch(b, e, d, a)
+	b.err = s.writeBatch(b, wEntryCnt|wAccessCnt|wIndexCnt)
 	close(b.c)
-	for e > s.capacity {
-		log.Trace("for >", "e", e, "s.capacity", s.capacity)
-		// Collect garbage in a separate goroutine
-		// to be able to interrupt this loop by s.quit.
-		done := make(chan struct{})
-		go func() {
-			s.collectGarbage(gcArrayFreeRatio)
-			log.Trace("collectGarbage closing done")
-			close(done)
-		}()
-
-		select {
-		case <-s.quit:
-			return errors.New("CollectGarbage terminated due to quit")
-		case <-done:
-		}
-		e = s.entryCnt
+	if s.entryCnt >= s.capacity {
+		go s.collectGarbage()
```

Comment:
now this will efffectively allow db capacity to be exceeded. 
For not long, but it is no longer a hard limit :)

Reply:
Right you are.

But judging from the diff it was like this before, too, it seems. Hmm.

Should we merely define a soft limit with a ratio of capacity? 5%, for example?

Reply:
Ah yes of course it was blocking ,good point.